### PR TITLE
Feature/skip submit

### PR DESCRIPTION
**Pull request by Behold3th**
Allows the creator to complete a bounty after the claim step.

This means that the bounty lifecycle has two options now: 

create --> claim --> submit --> complete
create --> claim  --> complete

This PR addresses the friction of having a contributor mark a bounty as `In-Review` if the bounty creator and bounty claimer engage in sync discussion over the bounty deliverable.

### CODE REVIEW

File: src/app/activity/bounty/Complete.ts [hunk 1]
```diff
@@ -58,7 +61,14 @@ export const completeBounty = async (request: CompleteRequest): Promise<void> =>
...
 	const bountyUrl = process.env.BOUNTY_BOARD_URL + request.bountyId;
-	const submittedByUser: GuildMember = await completedByUser.guild.members.fetch(getDbResult.dbBountyResult.submittedBy.discordId);
+	
+	let submittedByUserId: string;
+	if (bountyCompletedFromInProgress) {
+		submittedByUserId = getDbResult.dbBountyResult.claimedBy.discordId;
+	} else {
+		submittedByUserId = getDbResult.dbBountyResult.submittedBy.discordId;
+	}
+	const submittedByUser = await completedByUser.guild.members.fetch(submittedByUserId);
```
Comment (jordaniza) [inline src/app/activity/bounty/Complete.ts:71]: Yep makes sense

File: src/app/activity/bounty/Submit.ts [hunk 2]
```diff
@@ -39,18 +41,26 @@ export const submitBounty = async (request: SubmitRequest): Promise<void> => {
...
 			LogUtils.logError(`could not find bounty ${request.bountyId} in channel ${channelId} in guild ${request.guildId}`, e);
 			throw new RuntimeError(e);
 		});
     } else {
-        bountyEmbedMessage = request.message;
+        submitterMessage = request.message;
     }
 
+	if (getDbResult.dbBountyResult.creatorMessage !== undefined) {
+		const bountyChannel: TextChannel = await createdByUser.client.channels.fetch(getDbResult.dbBountyResult.creatorMessage.channelId) as TextChannel;
```
Comment (jordaniza) [inline src/app/activity/bounty/Submit.ts:53]: You don't need to cast and declare a variable. The cast is sufficient if you need to coerce the type.

File: src/app/validation/commandValidation.ts [hunk 3]
```diff
@@ -315,7 +315,10 @@ const complete = async (request: CompleteRequest): Promise<void> => {
             'Check your previous DMs from bountybot for the correct id.')
     }
 
-    if (dbBountyResult.status && dbBountyResult.status !== BountyStatus.in_review) {
+    const inReview = dbBountyResult.status && dbBountyResult.status === BountyStatus.in_review;
+    const inProgress = dbBountyResult.status && dbBountyResult.status === BountyStatus.in_progress;
+
```
Comment (jordaniza) [inline src/app/validation/commandValidation.ts:320]: What happens in the vase of !dbBountyResult.status

**Review by jordaniza — APPROVED** [review]
Lgtm, sensible PR

File: src/app/validation/commandValidation.ts (hunk 3 above)
Comment (Behold3th) [reply src/app/validation/commandValidation.ts:320]: Runtime error handling - if status isn't set, there's something seriously wrong

**Pull request merged**